library/Notifications/Widget/Calendar/BaseGrid.php

```diff
@@ -289,16 +340,92 @@ protected function assembleGridOverlay(BaseHtmlElement $overlay): void
                         'data-col-end' => $gridArea[3]
                     ])
                 );
-                $this->assembleEntry($entryHtml, $entry, $continuation);
+
+                $this->assembleEntry($entryHtml, $entry, $startText, $endText, $continuation);
                 $overlay->addHtml($entryHtml);
 
                 $continuation = true;
```

Comment:
Doesn't this already help in deciding whether a gradient is required or not? The only missing thing here, is that an entry which starts outside the current view isn't handled. This only covers a single direction, forward, and needs a counterpart. I didn't check how many of the calculations above are redundant, but since you're not using nor replacing this flag, I think some of them are.